packages/components/src/form-token-field/test/index.tsx

```diff
@@ -0,0 +1,2089 @@
+/**
+ * External dependencies
+ */
+import {
+	render,
+	screen,
+	within,
+	getDefaultNormalizer,
+	waitFor,
+} from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import type { ComponentProps } from 'react';
+
+/**
+ * WordPress dependencies
+ */
+import { useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import FormTokenField from '../';
+
+const FormTokenFieldWithState = ( {
+	onChange,
+	value,
+	initialValue = [],
+	...props
+}: ComponentProps< typeof FormTokenField > & {
+	initialValue?: ComponentProps< typeof FormTokenField >[ 'value' ];
+} ) => {
+	const [ selectedValue, setSelectedValue ] =
+		useState< ComponentProps< typeof FormTokenField >[ 'value' ] >(
+			initialValue
+		);
+
+	return (
+		<FormTokenField
+			{ ...props }
+			value={ selectedValue }
+			onChange={ ( tokens ) => {
+				setSelectedValue( tokens );
+				onChange?.( tokens );
+			} }
+		/>
+	);
+};
+
+const expectTokensToBeInTheDocument = ( tokensText: string[] ) => {
+	tokensText.forEach( ( tokenText, tokenIndex, tokensArray ) => {
+		// Each token has 2 tags rendered in the DOM:
+		// - one with the format "takenName (X of Y)", which is visibly hidden,
+		//   and is used for assistive technology;
+		// - one with the format "tokenName", which is visible but hidden to
+		//   assistive technology.
+		const assistiveTechnologyToken = screen.getByText(
+			`${ tokenText } (${ tokenIndex + 1 } of ${ tokensArray.length })`,
+			{
+				normalizer: getDefaultNormalizer( {
+					collapseWhitespace: false,
+					trim: false,
+				} ),
+			}
+		);
+		// The "exact" flag is necessary in order no to match the element
+		//  used for assistive technology.
+		const visibleToken = screen.getByText( tokenText, {
+			exact: true,
+			normalizer: getDefaultNormalizer( {
+				collapseWhitespace: false,
+				trim: false,
+			} ),
+		} );
+
+		expect( assistiveTechnologyToken ).toBeInTheDocument();
+		expect( visibleToken ).toBeInTheDocument();
```

Comment:
Should this be `toBeVisible()` instead, considering that this one is visible?

```suggestion
		expect( visibleToken ).toBeVisible();
```